includes/payment-methods/class-wc-stripe-payment-request.php

```diff
@@ -312,7 +312,13 @@ public function get_product_data() {
 		$product = wc_get_product( $post->ID );
 
 		if ( 'variable' === $product->get_type() ) {
-			$attributes = wc_clean( wp_unslash( $_GET ) );
+			$variation_attributes = $product->get_variation_attributes();
+			$attributes           = [];
+
+			foreach ( $variation_attributes as $attribute_name => $attribute_values ) {
+				$attribute_key                = 'attribute_' . sanitize_title( $attribute_name );
+				$attributes[ $attribute_key ] = isset( $_GET[ $attribute_key ] ) ? wc_clean( wp_unslash( $_GET[ $attribute_key ] ) ) : $product->get_variation_default_attribute( $attribute_name );
```

Comment:
This line is pretty long (though not at all the longest in the file) so I'd vote to split the ternary operands into separate lines, and/or add a code comment with a more concise description of the logic here.

Reply:
done https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1533/commits/4bbb710ab09d7f93e63c76c3ecffafff320a8841